clients/go/pkg/worker/jobWorker_builder.go

```diff
@@ -35,52 +35,57 @@ const (
 	DefaultRequestTimeout         = 10 * time.Second
 )
 
+var defaultBackoffSupplier = NewExponentialBackoffBuilder().Build()
+
 type JobWorkerBuilder struct {
 	gatewayClient  pb.GatewayClient
 	jobClient      JobClient
 	request        *pb.ActivateJobsRequest
 	requestTimeout time.Duration
 
-	handler       JobHandler
-	maxJobsActive int
-	concurrency   int
-	pollInterval  time.Duration
-	pollThreshold float64
-	metrics       JobWorkerMetrics
-	shouldRetry   func(context.Context, error) bool
+	handler         JobHandler
+	maxJobsActive   int
+	concurrency     int
+	pollInterval    time.Duration
+	pollThreshold   float64
+	metrics         JobWorkerMetrics
+	shouldRetry     func(context.Context, error) bool
+	backoffSupplier BackoffSupplier
 }
 
 type JobWorkerBuilderStep1 interface {
-	// Set the type of jobs to work on
+	// JobType Set the type of jobs to work on
 	JobType(string) JobWorkerBuilderStep2
 }
 
 type JobWorkerBuilderStep2 interface {
-	// Set the handler to process jobs. The worker should complete or fail the job. The handler implementation
+	// Handler Set the handler to process jobs. The worker should complete or fail the job. The handler implementation
 	// must be thread-safe.
 	Handler(JobHandler) JobWorkerBuilderStep3
 }
 
 type JobWorkerBuilderStep3 interface {
-	// Set the name of the worker owner
+	// Name Set the name of the worker owner
```

Comment:
This is idomatic go style for the comments. Let me know if I should remove them from the PR. 

There is no real need to change it now though.